pkg/component/delete/delete_test.go

```diff
@@ -0,0 +1,461 @@
+					client.EXPECT().GetOnePodFromSelector(selector).Return(nil, kerrors.NewForbidden(schema.GroupResource{}, "", nil))
+
+					labels := componentlabels.GetLabels(componentName, "app", false)
+					client.EXPECT().Delete(labels, false).Return(nil)
+					return client
+				},
+			},
+			args: args{
+				devfileObj:    odoTestingUtil.GetTestDevfileObj(fs),
+				componentName: componentName,
+			},
+			wantErr: false,
+		},
+		{
+			name: "failed to delete the component because of an un-ignorable error",
+			fields: fields{
+				kubeClient: func(ctrl *gomock.Controller, componentName string) kclient.ClientInterface {
+					client := kclient.NewMockClientInterface(ctrl)
+					selector := componentlabels.GetSelector(componentName, "app")
+					client.EXPECT().GetOnePodFromSelector(selector).Return(nil, errors.New("some un-ignorable error"))
+					return client
+				},
+			},
+			args: args{
+				devfileObj:    odoTestingUtil.GetTestDevfileObj(fs),
+				componentName: componentName,
+			},
+			wantErr: true,
+		},
+		{
+			name: "component deleted successfully with a devfile that has PreStopEvents",
+			fields: fields{
+				kubeClient: func(ctrl *gomock.Controller, componentName string) kclient.ClientInterface {
+					client := kclient.NewMockClientInterface(ctrl)
+					selector := componentlabels.GetSelector(componentName, "app")
+					client.EXPECT().GetOnePodFromSelector(selector).Return(odoTestingUtil.CreateFakePod(componentName, "runtime"), nil)
+					cmd := []string{"/bin/sh", "-c", "cd /projects/nodejs-starter && echo \"Hello World!\""}
+					client.EXPECT().ExecCMDInContainer("runtime", "runtime", cmd, gomock.Any(), gomock.Any(), nil, false).Return(nil)
+					labels := componentlabels.GetLabels(componentName, "app", false)
+					client.EXPECT().Delete(labels, false).Return(nil)
+					return client
+				},
+			},
+			args: args{
+				devfileObj:    odoTestingUtil.GetTestDevfileObjWithPreStopEvents(fs, "runtime", "echo \"Hello World!\""),
+				componentName: componentName,
+			},
+			wantErr: false,
+		},
+		{
+			name: "failed to delete component where a devfile that has PreStopEvents because pod is not in the running state",
+			fields: fields{
+				kubeClient: func(ctrl *gomock.Controller, componentName string) kclient.ClientInterface {
+					client := kclient.NewMockClientInterface(ctrl)
+					selector := componentlabels.GetSelector(componentName, "app")
+					pod := odoTestingUtil.CreateFakePod(componentName, "runtime")
+					pod.Status.Phase = corev1.PodFailed
+					client.EXPECT().GetOnePodFromSelector(selector).Return(pod, nil)
+					return client
+				},
+			},
+			args: args{
+				devfileObj:    odoTestingUtil.GetTestDevfileObjWithPreStopEvents(fs, "runtime", "echo \"Hello World!\""),
+				componentName: componentName,
+			},
+			wantErr: true,
+		},
+		{
+			name: "failed to delete component where a devfile that has PreStopEvents because it failed to execute the command inside the container",
```

Comment:
@kadel do we want this behaviour, or do we want to delete the component regardless of the reult of the preStopEvents commands?

Reply:
we definitely want to delete the component even if pre stop even fails. more info in https://github.com/redhat-developer/odo/pull/5500#discussion_r824572155